src/ray/raylet/scheduling_resources.cc

```diff
@@ -92,6 +98,29 @@ bool ResourceSet::SubtractResourcesStrict(const ResourceSet &other) {
   return !oversubscribed;
 }
 
+
+bool ResourceSet::SubtractResourcesStrict(const ResourceSet &other, bool delete_zero_capacity) {
+  // Subtract the resources and track whether a resource goes below zero.
+  bool oversubscribed = false;
+  for (const auto &resource_pair : other.GetResourceMap()) {
+    const std::string &resource_label = resource_pair.first;
+    const double &resource_capacity = resource_pair.second;
+    RAY_CHECK(resource_capacity_.count(resource_label) == 1)
```

Comment:
Is `resource_capacity` guaranteed to be positive? If so, include a check. If not, then it's ok for this check to fail.

Reply:
This function was an overload of `SubtractResourcesStrict(const ResourceSet &other)`. I have merged the two functions into one `SubtractResourcesStrict(const ResourceSet &other, bool delete_zero_capacity)` and replaced the old calls to `SubtractResourcesStrict(res_set)` with   `SubtractResourcesStrict(res_set, false)` in a new commit.

This check existed in the other function too so I didn't touch it. Should we remove it? I think it's a good sanity check to have to before subtracting resources.